network/gossip/libp2p/network.go

```diff
@@ -157,14 +165,22 @@ func (n *Network) Identity() (map[flow.Identifier]flow.Identity, error) {
 
 // Topology returns the identities of a uniform subset of nodes in protocol state using the topology provided earlier
 func (n *Network) Topology() (map[flow.Identifier]flow.Identity, error) {
-	subset, err := n.top.Subset(n.ids, n.fanout())
-	if err != nil {
-		return nil, fmt.Errorf("failed to derive list of peer nodes to connect to: %w", err)
+	myTopics := engine2.GetTopicsByRole(n.role)
+	var myFanout flow.IdentityList
+
+	// samples a connected component fanout from each topic and takes the
+	// union of all fanouts.
+	for _, topic := range myTopics {
+		subset, err := n.top.Subset(n.ids, n.fanout(), topic)
+		if err != nil {
+			return nil, fmt.Errorf("failed to derive list of peer nodes to connect for topic %s: %w", topic, err)
+		}
+		myFanout = append(myFanout, subset.Filter(filter.Not(filter.In(myFanout)))...)
```

Comment:
jordan introduced a new method - `Union` on `flow.IdentityList`. You can replace the `append` with that method.

Reply:
Fixed in https://github.com/onflow/flow-go/pull/83/commits/f6cf7b4ca0dfd0782e172a8baf77e983616254d5.